core/src/main/java/com/radixdlt/api/system/health/HealthInfoServiceImpl.java

```diff
@@ -106,4 +122,46 @@ public RecentSelfProposalMissStatistic recentSelfProposalMissStatistic() {
   public ProtocolState protocolState() {
     return inMemorySystemInfo.getProtocolState();
   }
+
+  @Override
+  public HealthResponse.ReadinessSignalStatusEnum readinessSignalStatus() {
+    return this.cachedReadinessSignalStatus;
+  }
+
+  @Override
+  public EventProcessor<HealthReadinessSignalStatusUpdateTrigger>
+      readinessSignalStatusUpdateTriggerEventProcessor() {
+    return unusedEvent ->
+        selfValidatorInfo
+            .bftValidatorId()
+            .ifPresent(
+                selfValidatorId -> {
+                  final var protocolState = inMemorySystemInfo.getProtocolState();
+                  final var signalRequired =
+                      protocolState.pendingProtocolUpdates().stream()
```

Comment:
It feels weird to me that this is per-protocol update, but then we flatten it to a single boolean, rather than per pending protocol update?

Perhaps instead we just cache our current vote? And then can compute the `readinessSignalStatus` for each pending update?

Reply:
I wanted to flatten it for the end user to a just a single status (VOTE_REQUIRED if there exists at least one pending protocol update that requires a vote)

Reply:
Hmm but if there is more than one protocol update it can never be true? Like you either be ready for one or the other? So it would always be VOTE_REQUIRED.

Also, VOTE_REQUIRED is old terminology, we should use the term readiness signalled.

So it doesn't seem to make sense really.

I think it makes sense instead to have a `readinessSignalStatus` for each pending protocol update, and either `NO_SIGNAL_REQUIRED` (if you're not a validator or the protocol update doesn't need a signal) or `READINESS_NOT_SIGNALLED` or `READINESS_SIGNALLED`.

Reply:
(done)